### ISSUE TITLE
Wrong click behavior in drilldown series when a chart is placed in a shadow root

### ISSUE THREAD

**Issue by DiegoCardoso** [highcharts/highcharts#23490]
I noticed that clicking in series with drilldown points does not always work as expected. Namely, every time the user drills up any level and then moves the cursor to a point with drilldown. Clicking on the point won't perform the drilldown, no matter how many times you click it. Only if you move the cursor out of the chart or onto another point and back, that clicking on the point work as intended.

While investigating it, my conclusion is that a set of factors is responsible for this issue:
- The accessibility module must be imported
- The chart is placed inside a shadow root
- Drilldown is enabled 

It seems that the issue comes from this line here:
https://github.com/highcharts/highcharts/blob/4f7b2a91f0ba68b9c531efe8c846e82bde8640eb/ts/Accessibility/KeyboardNavigation.ts#L314

As the `mouseup` event is attached to the document, the `target` property will reference the parent of the shadow root where the chart is rendered, instead of the SVG element from which the event comes. A possible fix would be to check the composed path to check for the presence of the element.

I created a reproduction of this issue here: https://jsfiddle.net/h1p2egqj/3/

To reproduce it, on the chart with the title "Shadow DOM":

1. Drill down on any of the points
2. In the breadcrumbs, click on "Things" to go back one level up
3. Move the cursor straight to any of the points and try to click on it (**does not work**)
4. To make it work, hover another point and then click on the first one tried

Note that while this was identified as an issue in drilldown series, it might affect use cases when charts + accessibility + shadow root are used.

**Comment by KacperMadej** [comment]
Hi @DiegoCardoso 

Thank you for reporting the bug.

_Internal note:_
When debugging, I noticed a difference sourcing from a11y - 

https://github.com/highcharts/highcharts/blob/0e89e3145a9704a5fcc6f92bab8eb68a3d15513f/ts/Accessibility/KeyboardNavigation.ts#L319
  
On mouse up the event is terminated, which fires `onMouseOut` for the series which nullifies hoverPoint preventing the `onContainerClick` from firing click on the point.
